bundle-workflow/src/test_workflow/perf_test_cluster.py

```diff
@@ -0,0 +1,63 @@
+import json
+import os
+import subprocess
+from contextlib import contextmanager
+
+from test_workflow.test_cluster import TestCluster
+
+
+class PerformanceTestCluster(TestCluster):
+    """
+    Represents a performance test cluster. This class deploys the opensearch bundle with CDK and returns the private IP.
+    """
+
+    def __init__(self, bundle_manifest, config, stack_name, security):
+        self.manifest = bundle_manifest
+        self.security_id = config['Constants']['SecurityGroupId']
+        self.vpc_id = config['Constants']['VpcId']
+        self.account_id = config['Constants']['AccountId']
+        self.region = config['Constants']['Region']
+        self.role = config['Constants']['Role']
+        self.work_dir = 'tools/cdk/mensor/single-node/'
+        self.stack_name = stack_name
+        self.cluster_endpoint = None
+        self.cluster_port = None
+        self.output_file = 'output.json'
+        self.ip_address = None
+        self.security = 'enable' if security else 'disable'
+        self.params = f'-c url={self.manifest.build.location} -c security_group_id={self.security_id} -c vpc_id={self.vpc_id}'\
+                      f' -c account_id={self.account_id} -c region={self.region} -c stack_name={self.stack_name} -c security={self.security}'\
+                      f' -c architecture={self.manifest.build.architecture} --require-approval=never --plugin cdk-assume-role-credential-plugin'\
+                      f' -c assume-role-credentials:writeIamRoleName={self.role} -c assume-role-credentials:readIamRoleName={self.role}'
+
+    @contextmanager
```

Comment:
Extract into a class.

Reply:
Should I create a separate class in the same file as this requirement is specific to perf test? Something like
`class Cluster:`
           `@contextmanager`
             `def __init__(self, manifest, config, stack_name, security):`
               ` perf_test_cluster = PerformanceTestCluster(manifest, config, stack_name, security)`

Reply:
Created a new class in the same file as this yield version is specific to perf test